### PR TITLE
[AIRFLOW-2868] Extend mesos_executor

### PR DESCRIPTION
**Pull request by amir656**
### Jira

- [x] My PR addresses the following [AIRFLOW-2868](https://issues.apache.org/jira/browse/AIRFLOW-2868) issue and references it in the PR title. 

### Description

- [x] I extended the mesos_executor to be able to execute different tasks with different CPU and Memory requirements, as well as different docker images. By providing an executor_config dictionary, users can specify these tasks in their operators. This builds off of the executor config created for the kubernetes_executor.

### Tests

- [x] My PR added to tests/contrib/executors/test_mesos_executor.py

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it. 
Please see comment below.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by amir656** [comment]
I am not sure where to add details about this change to the documentation. There doesn't seem to be documentation about the executor_config, or even the kubernetes executor at all, which I would have hoped to extend. Please let me know if you have any ideas about where to add documentation

**Comment by ali5h** [comment]
fix the commit and force push, it is not pointing to ur github account.

File: airflow/contrib/executors/mesos_executor.py [hunk 1]
```diff
@@ -238,15 +290,15 @@ def start(self):
...
+        task_cpu = configuration.conf.getint('mesos', 'TASK_CPU')
+        task_mem = configuration.conf.getint('mesos', 'TASK_MEMORY')
 
-        if not configuration.conf.get('mesos', 'TASK_MEMORY'):
-            task_memory = 256
-        else:
-            task_memory = configuration.conf.getint('mesos', 'TASK_MEMORY')
+        docker_image = None
+        # Configuration errors when DOCKER_IMAGE_SLAVE is not present in config
+        if 'DOCKER_IMAGE_SLAVE' in configuration.as_dict()['mesos']:
```
Comment (ali5h) [inline airflow/contrib/executors/mesos_executor.py:298]: `try: catch:` ?

**Comment by ron819** [comment]
@amir656 are yiy still working on this PR?

**Comment by amir656** [comment]
@ron819 I haven't worked on it in a while because school started up again. But I'm done with finals on Wednesday, so I will try and finish this up after that. Do you have any suggestions for where I could add to the documentation?

**Comment by feng-tao** [comment]
PTAL @ashb

**Comment by feng-tao** [comment]
@amir656 do you want to rebase your pr with the master?

File: airflow/contrib/executors/mesos_executor.py (hunk 1 above)
Comment (ashb) [reply airflow/contrib/executors/mesos_executor.py:298]: `as_dict()` on the whole airflow config object is relatively expensive, and should be avoided.

```suggestion
        if configuration.has_option('mesos', 'docker_image_slave'):
```

Additionally: Please don't use the term "slave" - "worker" or "executor" or "task" would be better terms here.

File: airflow/www/app.py [hunk 2]
```diff
@@ -193,7 +193,7 @@ def shutdown_session(exception=None):
 
 
 def root_app(env, resp):
-    resp(b'404 Not Found', [(b'Content-Type', b'text/plain')])
+    resp(b'404 Not Found', [('Content-Type', 'text/plain')])
```
Comment (ashb) [inline airflow/www/app.py]: Unrelated change to Mesos. Remove.

File: airflow/www/security.py [hunk 3]
```diff
@@ -18,7 +18,6 @@
 # under the License.
 #
 
-import logging
```
Comment (ashb) [inline airflow/www/security.py]: Unrelated change to Mesos. Remove any changes from this file.

File: UPDATING.md [hunk 4]
```diff
@@ -107,14 +107,6 @@ so you might need to update your config.
 
 `task_runner = StandardTaskRunner`
 
-### DAG level Access Control for new RBAC UI
```
Comment (ashb) [inline UPDATING.md]: Unrelated change to Mesos. Remove all changes in this file.

**Review by ashb — CHANGES_REQUESTED** [review]

**Comment by OmerJog** [comment]
@kaxil This can be closed due to https://github.com/apache/airflow/pull/5115

**Comment by ashb** [comment]
The Airflow community is deprecating the Mesos Executor as it is under-supported and has been broken for a few releases but no one complained (oops!)

If someone wants to pick up support for it I can explain how to pull it out into a third-party module.
